crystal-map-connector/src/main/kotlin/ch/ergon/dope/extension/CaseExpression.kt

```diff
@@ -0,0 +1,53 @@
+package ch.ergon.dope.extension
+
+import ch.ergon.dope.resolvable.WhenThenCondition
+import ch.ergon.dope.resolvable.expression.Expression
+import ch.ergon.dope.resolvable.expression.case
+import ch.ergon.dope.toDopeType
+import com.schwarz.crystalapi.schema.CMType
+
+fun case(
+    expression: CMType,
+    whenThenCondition: WhenThenCondition,
+    vararg additionalWhenThenConditions: WhenThenCondition,
+    elseCase: Expression? = null,
+) = case(
+    expression = expression.toDopeType(),
+    whenThenCondition = whenThenCondition,
+    additionalWhenThenConditions = additionalWhenThenConditions,
+    elseCase = elseCase,
+)
+
+fun case(
+    expression: Expression,
+    whenThenCondition: WhenThenCondition,
+    vararg additionalWhenThenConditions: WhenThenCondition,
+    elseCase: CMType? = null,
+) = case(
+    expression = expression,
+    whenThenCondition = whenThenCondition,
+    additionalWhenThenConditions = additionalWhenThenConditions,
+    elseCase = elseCase?.toDopeType(),
+)
+
+fun case(
+    whenThenCondition: WhenThenCondition,
+    vararg additionalWhenThenConditions: WhenThenCondition,
+    elseCase: CMType? = null,
+) = case(
+    whenThenCondition = whenThenCondition,
+    additionalWhenThenConditions = additionalWhenThenConditions,
+    elseCase = elseCase?.toDopeType(),
+)
+
+fun case(
+    expression: CMType? = null,
```

Comment:
Why did you make this nullable? If I understood correctly, the extension above already covers the searched case expression.

Reply:
removed it